ostd/src/arch/riscv/device/io_port.rs

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! I/O port access.
+
+use core::marker::PhantomData;
+
+pub struct WriteOnlyAccess;
+pub struct ReadWriteAccess;
+
+pub trait IoPortWriteAccess {}
+pub trait IoPortReadAccess {}
+
+impl IoPortWriteAccess for WriteOnlyAccess {}
+impl IoPortWriteAccess for ReadWriteAccess {}
+impl IoPortReadAccess for ReadWriteAccess {}
+
+pub trait PortRead: Sized {
+    unsafe fn read_from_port(_port: u16) -> Self {
+        unimplemented!()
```

Comment:
A dummy `IoPort` type is implemented because (1) RISC-V supports and needs port-based I/O, or (2) our kernel's CPU architecture-independent code needs the `IoPort` type to compile?

Reply:
Let's leave this issue to future PRs. We can't make everything right in a single PR.